scripts/ci-build.sh

```diff
@@ -0,0 +1,562 @@
+    echo "Fetching from $value"
+    git remote add temp_fork "$value"
+    git fetch temp_fork
+
+    # Master is assumed, but it is likely overridden by a COMMIT/TAG/BRANCH
+    # value below. but this could fail.
+    echo "NOTE: Assuming master branch for now (might be overridden later)"
+    echo "NOTE: This will fail if no master branch exists in $value"
+    echo "+ git reset temp_fork/master --hard"
+    git reset temp_fork/master --hard
+    cd "$d"
+  fi
+
+  # ...then the others
+  for suffix in COMMIT TAG BRANCH ; do
+    var_name="LAYER_${clean_name}_${suffix}"
+    value=$(deref "$var_name")
+    if [ -n "$value" ] ; then
+      echo "***** NOTE: OVERRIDING LAYER $name USING \$${var_name} = $value *****"
+      d="$PWD"
+      cd $name || echo "Layer name wrong?  Can't cd to dir : $name"
+      git fetch --all
+
+      # If forked, make sure to use the right remote, otherwise origin
+      # For branches, prefer the explicit: remote/branchname
+      if git remote | grep -q temp_fork ; then
+        git checkout "temp_fork/$value" 2>/dev/null || git checkout $value
+      else
+        git checkout "origin/$value" 2>/dev/null || git checkout $value
+      fi
+
+      cd "$d"
+      break  # First one wins, priority order: COMMIT >= TAG >= BRANCH
+    fi
+  done
+}
+
+# Overrides are defined invisibly in environment variables so we should inform
+# the user which ones are applied, for both interactive use and for studying
+# a build log at a later time.
+# NOTE:  Because of the order they are called we can't use output from the
+# layer_override function above, so we have some repeated code here.
+print_layer_overrides() {
+  local names="$1" name clean_name suffix var_name value d
+
+  for name in $names ; do
+    clean_name=$(echo "$name" | sed 's/-/_/g')
+    for suffix in COMMIT TAG BRANCH ; do
+      var_name="LAYER_${clean_name}_${suffix}"
+      value=$(deref "$var_name")
+      if [ -n "$value" ] ; then
+        echo "$var_name = $value"
+      fi
+    done
+  done
+}
+
+# List all dependent layers here.  This list is used by the layer overrides
+# functionality.
+# NOTE If a super-project is applied, such as using submodules, then this list
+# would be fetched from git instead, in case it changes.
+LAYERS="
+poky
+meta-gplv2
+meta-openembedded
+renesas-rcar-gen3
+"
+
+# Clean up function is called at end of script, or as a signal handler (trap)
+# if the script is interrupted
+cleanup() {
+  # Restore git config user - if this was not defined locally before then it is
+  # unset (which might mean a global setting is used)
+  if [ -z "$olduser" ] ; then
+    git config --unset user.name
+  else
+    git config user.name "$olduser"
+  fi
+  if [ -z "$oldemail" ] ; then
+    git config --unset user.email
+  else
+    git config user.email "$oldemail"
+  fi
+}
+
+# ---- Main program ----
+
+trap cleanup SIGINT SIGTERM
+
+D=$(dirname "$0")
+cd "$D"
+BASEDIR=$(readlink -f "$PWD/..")
```

Comment:
For some reason we wind up building in <path>/meta-ivi/build/... I'd prefer to keep the build directory in <path>/build, the same level as the other meta-layers.